src/prog_models/prognostics_model.py

```diff
@@ -1113,21 +1126,76 @@ def calc_error(self, times: List[float], inputs: List[InputContainer], outputs:
+        
+        # Ensure error methods provide correct results via hand-calculating each one and equating each one.
+        acceptable_types = {Sequence, np.ndarray, SimResult, LazySimResult}
+
+        if not all(isinstance(obj, tuple(acceptable_types)) for obj in [times, inputs, outputs]):
+            type_error = f"Types passed in must be from the following: Sequence, np.ndarray, SimResult, or LazySimResult. Current types" \
+                         f"{(' at data location (' + str(_loc) + ')' if _loc is not None else '')}" \
+                         f": times = {type(times).__name__}, inputs = {type(inputs).__name__}, and outputs = {type(outputs).__name__}."
+            raise TypeError(type_error)
+        if len(times) != len(inputs) or len(inputs) != len(outputs):
+            len_error = f"Times, inputs, and outputs must all be the same length. Current lengths" \
+                        f"{(' at data location (' + str(_loc) + ')' if _loc is not None else '')}" \
+                        f": times = {len(times)}, inputs = {len(inputs)}, outputs = {len(outputs)}."
+            raise ValueError(len_error)
+        if len(times) < 2:
+            less_2_error = f"Must provide at least 2 data points for times, inputs, and outputs" \
+                           f"{(' at data location (' + str(_loc) + ')' if _loc is not None else '')}."
+            raise ValueError(less_2_error)
+
+        # Determines if all values of arguments are iterables
+        self.check_iterable(times, 'times', loc=_loc) if _loc is not None else self.check_iterable(times, 'times')
+        self.check_iterable(inputs, 'inputs', loc=_loc) if _loc is not None else self.check_iterable(inputs, 'inputs')
+        self.check_iterable(outputs, 'outputs', loc=_loc) if _loc is not None else self.check_iterable(outputs, 'outputs')
+
+        if isinstance(times[0], str):
+            raise TypeError("Times values cannot be strings")
+        if isinstance(times[0], Iterable):
+            # Calculate error for each
+            error = []
+            for r, (t, i, z) in enumerate(zip(times, inputs, outputs)):
+                run_updated = str(r) if _loc is None else _loc + f', {str(r)}'
+                error.append(self.calc_error(t, i, z, _loc=run_updated, **kwargs))
+            return sum(error)/len(error)
+            
+        dt = kwargs.get('dt', 1e99)
+        kwargs['stability_tol'] = kwargs.get('stability_tol', 0.95)
+
+        # Checks stability_tol is within bounds
+        # Throwing a default after the warning.
+        if not isinstance(kwargs['stability_tol'], Number):
+            raise TypeError(f"Keyword argument 'stability_tol' must be either a int, float, or double.")
+        if kwargs['stability_tol'] > 1 or kwargs['stability_tol'] <= 0:
+            warn(f"Configurable cutoff must be some float value in the domain (0, 1]. Received {kwargs['stability_tol']}. Resetting value to 0.95.")
+            kwargs['stability_tol'] = 0.95
```

Comment:
Let's change this to an exception

Reply:
I like this idea. It solves our previous issue of needing to re-define kwargs or stability_tol